packages/js-commons/src/local-files.ts

```diff
@@ -0,0 +1,380 @@
+import { join } from "node:path";
+import { exec } from "node:child_process";
+import { existsSync, mkdirSync } from "node:fs";
+import { readFile, writeFile } from "node:fs/promises";
+import type { LocalApplicationTransporter } from "./transporter";
+import {
+  type ThemeDTO,
+  type AssetDTO,
+  type ScreenDTO,
+  type WidgetDTO,
+  type ScriptDTO,
+  type TranslationDTO,
+  type ApplicationDTO,
+  type EnvironmentDTO,
+  type EnsembleDocument,
+  type ApplicationMetaDTO,
+  EnsembleDocumentType,
+} from "./dto";
+
+export const EnsembleHiddenFolder = ".ensemble";
+export const HiddenManifestFile = ".manifest.json";
+export const AppManifestDataFile = "apps-manifest.json";
+
+export const ArtifactYamlFolderNames = {
+  theme: "Theme",
+  screens: "Screens",
+  widgets: "Widgets",
+  scripts: "Scripts",
+  translations: "Translations",
+} as const;
+
+export const ArtifactFilesName = {
+  theme: "theme.json",
+  assets: "asset.json",
+  screens: "screen.json",
+  widgets: "widget.json",
+  scripts: "script.json",
+  translations: "i18n.json",
+  env: "environment.json",
+  groupLabels: "label.json",
+} as const;
+
+export const getLocalApplicationTransporter = (
+  yamlFolder: string,
+  globalFolder: string,
+): LocalApplicationTransporter => ({
+  get: async (appId: string): Promise<ApplicationDTO> => {
+    const userYamlAppFolder = await getYamlFolderPath(
+      appId,
+      yamlFolder,
+      globalFolder,
+    );
+    const userAppFolder = join(userYamlAppFolder, EnsembleHiddenFolder, appId);
+    if (!existsSync(userAppFolder))
+      throw Error("App data not found on user local disk.");
+
+    const [
+      env,
+      theme,
+      assets,
+      screens,
+      widgets,
+      scripts,
+      groupLabels,
+      translations,
+    ] = await Promise.all([
+      readJsonFile<EnvironmentDTO>(join(userAppFolder, ArtifactFilesName.env)),
+      readJsonFile<ThemeDTO>(join(userAppFolder, ArtifactFilesName.theme)),
+      readJsonFile<AssetDTO[]>(join(userAppFolder, ArtifactFilesName.assets)),
+      readJsonFile<ScreenDTO[]>(join(userAppFolder, ArtifactFilesName.screens)),
+      readJsonFile<WidgetDTO[]>(join(userAppFolder, ArtifactFilesName.widgets)),
+      readJsonFile<ScriptDTO[]>(join(userAppFolder, ArtifactFilesName.scripts)),
+      readJsonFile<Record<string, string>>(
+        join(userAppFolder, ArtifactFilesName.groupLabels),
+      ),
+      readJsonFile<TranslationDTO[]>(
+        join(userAppFolder, ArtifactFilesName.translations),
+      ),
```

Comment:
Why did you decide to break this metadata up into individual subdirectories? This can all be stored in one manifest file at the hidden app folder level. Opening up multiple file handles consumes more memory than necessary.